### ISSUE TITLE
TestInterp1D.test_read_only test failure

### ISSUE THREAD

**Issue by rgommers** [scipy/scipy#9464]
On current master (at 1.2.x branch point), linux, Cython 0.27.3, gcc 8.2.1. Test recently introduced, in gh-9350.
```
____________________________ TestInterp1D.test_read_only _____________________________

self = <scipy.interpolate.tests.test_interpolate.TestInterp1D object at 0x7f66cdaa7cf8>

    def test_read_only(self):
        x = np.arange(0, 10)
        y = np.exp(-x / 3.0)
        xnew = np.arange(0, 9, 0.1)
        # Check both read-only and not read-only:
        for writeable in (True, False):
            xnew.flags.writeable = writeable
            for kind in ('linear', 'nearest', 'zero', 'slinear', 'quadratic',
                         'cubic'):
                f = interp1d(x, y, kind=kind)
>               vals = f(xnew)

f          = <scipy.interpolate.interpolate.interp1d object at 0x7f66cdab0c28>
kind       = 'zero'
self       = <scipy.interpolate.tests.test_interpolate.TestInterp1D object at 0x7f66cdaa7cf8>
vals       = array([1.        , 1.        , 1.        , 1.        , 1.        ,
       1.        , 0.71653131, 0.71653131, 0.716531...45, 0.06948345, 0.06948345, 0.06948345, 0.06948345,
       0.06948345, 0.04978707, 0.04978707, 0.04978707, 0.04978707])
writeable  = False
x          = array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
xnew       = array([0. , 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1. , 1.1, 1.2,
       1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9, 2. ,..., 6.8, 6.9, 7. , 7.1, 7.2, 7.3, 7.4, 7.5, 7.6, 7.7,
       7.8, 7.9, 8. , 8.1, 8.2, 8.3, 8.4, 8.5, 8.6, 8.7, 8.8, 8.9])
y          = array([1.        , 0.71653131, 0.51341712, 0.36787944, 0.26359714,
       0.1888756 , 0.13533528, 0.09697197, 0.06948345, 0.04978707])

../scipy/scipy/interpolate/tests/test_interpolate.py:711: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../scipy/scipy/interpolate/polyint.py:79: in __call__
    y = self._evaluate(x)
../scipy/scipy/interpolate/interpolate.py:662: in _evaluate
    y_new = self._call(self, x_new)
../scipy/scipy/interpolate/interpolate.py:650: in _call_spline
    return self._spline(x_new)
../scipy/scipy/interpolate/_bsplines.py:348: in __call__
    self._evaluate(x, nu, extrapolate, out)
../scipy/scipy/interpolate/_bsplines.py:359: in _evaluate
    self.k, xp, nu, extrapolate, out)
_bspl.pyx:95: in scipy.interpolate._bspl.evaluate_spline
    ???
stringsource:646: in View.MemoryView.memoryview_cwrapper
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   ValueError: buffer source array is read-only

__builtins__ = <builtins>
__doc__    = '\nRoutines for evaluating and manipulating B-splines.\n\n'
__file__   = '/home/rgommers/code/scipy/scipy/interpolate/_bspl.cpython-36m-x86_64-linux-gnu.so'
__loader__ = <_frozen_importlib_external.ExtensionFileLoader object at 0x7f66bc830be0>
__name__   = 'scipy.interpolate._bspl'
__package__ = 'scipy.interpolate'
__pyx_unpickle_Enum = <built-in function __pyx_unpickle_Enum>
__spec__   = ModuleSpec(name='scipy.interpolate._bspl', loader=<_frozen_importlib_external.ExtensionFileLoader object at 0x7f66bc830be0>, origin='/home/rgommers/code/scipy/scipy/interpolate/_bspl.cpython-36m-x86_64-linux-gnu.so')
__test__   = {'evaluate_all_bspl (line 163)': "Evaluate the ``k+1`` B-splines which are non-zero on interval ``m``.\n\n    Paramete..., c[i] + '--', lw=3, label=str(i))\n    ...\n    >>> plt.grid(True)\n    >>> plt.legend()\n    >>> plt.show()\n\n    "}
_colloc    = <built-in function _colloc>
_handle_lhs_derivatives = <built-in function _handle_lhs_derivatives>
_norm_eq_lsq = <cyfunction _norm_eq_lsq at 0x7f66bc9d01e8>
evaluate_all_bspl = <built-in function evaluate_all_bspl>
evaluate_spline = <cyfunction evaluate_spline at 0x7f66bce80ee8>
np         = <module 'numpy' from '/home/rgommers/anaconda3/lib/python3.6/site-packages/numpy/__init__.py'>

stringsource:347: ValueError
```

**Comment by ev-br** [comment]
Upgrade cython to at least 0.28.x?

**Comment by ilayn** [comment]
We had some issues with Cython 0.28 versions in the past #8581 and #8562 and possibly other ones in the related issues

**Comment by rgommers** [comment]
The 0.28 issues seems fixed now, and 0.29 makes this failure go away. So bumping the minimum Cython version to either of those is fine with me.

**Comment by charris** [comment]
Cython  also caused problems with OpenBSD, need >= 0.28.5. See https://github.com/numpy/numpy/issues/11664.

**Comment by rgommers** [comment]
Ah yes, thanks @charris. Looks like we should be requiring >= 0.28.5 then.

**Comment by charris** [comment]
@tylerjereddy Note that you will need recent Cython for the source distributions.

**Comment by tylerjereddy** [comment]
Noted, thanks

**Comment by rgommers** [comment]
should be fixed in 1.2.0 by using Cython 0.29.0, closing

**Comment by dmadeka** [comment]
Does this also work for CubicSplines? Or is it only applicable to `interp1d`?
